src/prob/tp_opf.jl

```diff
@@ -33,6 +33,7 @@ function post_tp_opf(pm::GenericPowerModel)
 
         for i in ids(pm, :ref_buses)
             constraint_tp_theta_ref(pm, i, ph=h)
+            constraint_tp_voltage_magnitude_ref(pm, i, ph=h)
```

Comment:
This looks like power flow semantics to me rather than optimal power flow semantics.  I know is is typical to fix the root of a feeder to 1.0 V p.u., but isn't it possible to vary that set point using the step-down transformer that is feeding it?

Reply:
There is a challenge related to recovering node voltage variables in the original variable space. when you only fix the angles at the reference bus. It may not be possible to recover a solution if W is not rank-1, and you need that as a starting point to run algorithm 1 or 2 in Gan and Low.

Gan and Low can be made to work with just fixing the voltage angles, if the voltage angles are 120 degrees and the voltage magnitudes are identical for the three phases. Your W matrix variable for W=U*U' with U = [um/_0, um/_120, um/_0, ] and w = um^2 will then look like
```
W = 
[ w, -w/2 -w/2;
-w/2, w, -w/2;
-w/2, -w/2, w] 
+ j * sqrt(3)/2*
[0, w, -w;
-w, 0, w;
w, -w, 0]
```

We'll have to work out other reference bus cases where the angles are not evenly spaced / magnitudes are not identical, but rank-1-ness may not easily be guaranteed.

Reply:
The issue of problem formulation and feasible solution recovery from a relaxation are orthogonal.  The standard OPF problem provides no guarantees of feasible solution recovery.  If you would like to propose an OPF variant, which has such properties, that is fine just provide a doc string explaining the motivation for the formulation and a citation if it makes sense.